homeassistant/components/overkiz/water_heater_entities/domestic_hot_water_production.py

```diff
@@ -276,6 +303,23 @@ async def async_set_operation_mode(self, operation_mode: str) -> None:
                         },
                     )
 
-        await self.executor.async_execute_command(
-            OverkizCommand.SET_DHW_MODE, OPERATION_MODE_TO_OVERKIZ[operation_mode]
+        dhw_mode = OPERATION_MODE_TO_OVERKIZ[operation_mode]
+
+        # Check if the dhw_mode we're trying to set is correctly enable for this device
+        state_mode_definition = self.executor.select_definition_state(
```

Comment:
I would prefer to do this in the `__init__` instead of at runtime.

Reply:
Done, the code now create a specific `self.overkiz_to_operation_mode` on `__init__`